internal/pkg/skuba/deployments/ssh/kubernetes.go

```diff
@@ -56,24 +58,99 @@ func kubernetesUploadSecrets(errorHandling KubernetesUploadSecretsErrorBehavior)
 	}
 }
 
-func kubernetesInstallNodePattern(t *Target, data interface{}) error {
+func kubernetesParseInterfaceVersions(data interface{}) (string, string, error) {
 	kubernetesBaseOSConfiguration, ok := data.(deployments.KubernetesBaseOSConfiguration)
 	if !ok {
-		return errors.New("couldn't access kubernetes base OS configuration")
+		return "", "", errors.New("couldn't access kubernetes base OS configuration")
 	}
 
 	v, err := version.ParseSemantic(kubernetesBaseOSConfiguration.CurrentVersion)
 	if err != nil {
-		return err
+		return "", "", err
 	}
 	currentVersion := kubernetes.MajorMinorVersion(v)
-	patternName := fmt.Sprintf("patterns-caasp-Node-%s", currentVersion)
+	updatedVersion := ""
 
 	if kubernetesBaseOSConfiguration.UpdatedVersion != "" {
-		updatedVersion := kubernetes.MajorMinorVersion(version.MustParseSemantic(kubernetesBaseOSConfiguration.UpdatedVersion))
-		patternName = fmt.Sprintf("patterns-caasp-Node-%s-%s", currentVersion, updatedVersion)
+		updatedVersion = kubernetes.MajorMinorVersion(version.MustParseSemantic(kubernetesBaseOSConfiguration.UpdatedVersion))
+	}
+	return currentVersion, updatedVersion, nil
+}
+
+func kubernetesFreshInstallAllPkgs(t *Target, data interface{}) error {
+	current, _, err := kubernetesParseInterfaceVersions(data)
+	if err != nil {
+		return err
+	}
+	var pkgs []string
+
+	// Standard packages for a new cluster
+	pkgs = append(pkgs, "+caasp-release", "skuba-update", "supportutils-plugin-suse-caasp", "cri-tools")
+	// Version specific
+	pkgs = append(pkgs, fmt.Sprintf("+kubernetes-%s-kubeadm", current))
+	pkgs = append(pkgs, fmt.Sprintf("+kubernetes-%s-kubelet", current))
+	pkgs = append(pkgs, fmt.Sprintf("+kubernetes-%s-client", current))
+	pkgs = append(pkgs, fmt.Sprintf("+cri-o-%s*", current))
```

Comment:
Maybe we could put these patterns inside of a constant list to reuse them later inside of the code (eg: `kubernetesUpgradeOtherPkgs`)?

Maybe two lists: `phase1pkgs` and `phase2kpgs` :thinking:

Reply:
I hesitated there, because I have the impression they will be fluctuating over time. But indeed, they aren't right now, so that might make sense.

Reply:
Let's keep that for now, if you don't mind.